### PR TITLE
MAINT: Extract string helpers to a new private file

### PR DESCRIPTION
**Pull request by eric-wieser**
A simple cut-and-paste. I think eventually we can just eliminate these functions from internal use.

### CODE REVIEW

File: numpy/core/_string_helpers.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+_ascii_upper = _all_chars[65:65+26]
+_ascii_lower = _all_chars[97:97+26]
+LOWER_TABLE = "".join(_all_chars[:65] + _ascii_lower + _all_chars[65+26:])
+UPPER_TABLE = "".join(_all_chars[:97] + _ascii_upper + _all_chars[97+26:])
+
+
+def english_lower(s):
+    """ Apply English case rules to convert ASCII strings to all lower case.
+
+    This is an internal utility function to replace calls to str.lower() such
```
Comment (eric-wieser) [inline numpy/core/_string_helpers.py]: Note that on python 2, `str.lower` is locale-independent anyway.

**Comment by mhvk** [comment]
Change itself looks fine, but what is the benefit of having this in a separate file?

**Comment by eric-wieser** [comment]
I'd like to try and split numerictypes into two pieces, "expose aliased type names" and "provide mostly-unused functions". The way we name scalar types is still a mess, and the smaller I can make the file that does it, the easier refactors will be.

**Review by mhvk — APPROVED** [review]
Changes are OK.

**Comment by mhvk** [comment]
OK, I can see how that would simplify refactoring. Merging...

**Pull request merged**